decision-process/README.md

```diff
@@ -0,0 +1,134 @@
+---
+creation_date: 2023-10-10
+issues:
+  - https://github.com/giantswarm/giantswarm/issues/24661
+owners:
+  - https://github.com/orgs/giantswarm/teams/sig-architecture
+  - https://github.com/orgs/giantswarm/teams/team-horizon
+  - https://github.com/orgs/giantswarm/teams/team-team
+state: review
+---
+
+# RFC and decision making process
+
+This RFC-of-RFCs aims to expand on the current RFC process by introducing an explicit, lightweight decision making process.
+
+We use the terms "RFC" and "decision" quite interchangeably. Think of a _decision_ as exactly what the word means, and an _RFC_ as its written-down version as Markdown file.
+
+At the moment, the concept of [ADRs](https://intranet.giantswarm.io/docs/product/architecture-specs-adrs/adr/) and [PDRs](https://intranet.giantswarm.io/docs/product/pdr/) are not covered here and their processes do not change yet. We may want to amend this later.
+
+Goals of the process:
+
+- Quick progress on RFCs and decision making
+- Relevant groups have seen and approved the decision
+- More structured RFC format to add rendering and further automation
+- Better overview by rendering RFCs
+- Replace the [rfc repository's README](https://github.com/giantswarm/rfc/blob/b3f4d2af6ff8965e542baae7d0e8fadb9163f356/README.md)
+- Giant Swarm folks using RFCs more readily and happily because they see that the process is working
+
+## RFC file structure
+
+To allow for a decision making process without removing the current free-form nature of RFCs, some structured information is proposed.
+
+The following changes will be introduced:
+
+- All RFCs will have a YAML header with structured information that can be parsed and used e.g. for rendering the RFCs in our [handbook](https://handbook.giantswarm.io/)
+- The YAML header must contain the following fields in this order:
+
+  - `creation_date: YYYY-MM-DD`
+  - `issues` containing links to GitHub issues. Can be null or an empty list. The issues can be the origin of the RFC or where the work for implementing the decision takes place.
+  - `owners` containing GitHub team URLs. This list must not be empty. Please enter only the team and/or SIG names of the author and owners, not the teams who should comment on the RFC (the _stakeholders_, for which we don't have a structured field). The `owners` field contains teams who have the expertise to check the RFC after a few months. It may later be used to automatically create GitHub issues "Please review if this RFC is still relevant" (see [Future ideas](#future-ideas)).
```

Comment:
i feel like stakeholders is somehow necessary? how can we make sure that relevant parties outside of owners have seen / accepted an RFC?

Reply:
I assume you are asking about getting a minimum number of approvals from relevant teams without forgetting important people – that must be represented as "reviewers" on the PR, since there's no reasonable way to sync `stakeholders: [ ... ]` and PR reviewers (too complex). The chance of forgetting a stakeholder in a YAML field is likely the same as forgetting to set a reviewer on the PR – and adding a team on the PR is just a few clicks and therefore much easier. If you later want to see who reviewed/approved/commented, we just look at the original PR. If there are complaints that a certain team wasn't involved, that's not really what a YAML field can heal, in my opinion, but it could be a mindset or laziness issue, for instance, missing knowledge of the author, or a plain human mistake. That's why I think we should have some very central place where RFCs get announced, and/or regularly discussed. The relevant SIG should take care of adding missing reviewers on the PR. ◀️ Should I add this clarification to the RFC? I agree about the concern and that we should solve it as part of this written process.

If you had concrete ideas about the YAML field, let me know. Maybe you had a good solution in mind?

Reply:
my thinking was having a field similar to owners, and then do some CI magic to ensure that at least one person out of the listed teams has approved

my hunch is that having it in the text means that people will consider it more, than if it's in github, but i'm not too sure on that

happy to go either way, i think

Reply:
I'd recommend we leave this out for now, and if it doesn't work, we can improve the flow by adding the required field

Reply:
My first gut feeling was like Joe's, that we should also have stakeholders, but then I thought that I prefer this RFC being as is right now, simpler.

Then after we we see how this simpler version is working in practice, we can have another RFC which makes stakeholders mandatory, so we can also have e.g. like Joe said CI checks if there is an approval from every stakeholder.